plugin/storage/es/mappings/jaeger-service-with-ilm-7.json

```diff
@@ -0,0 +1,49 @@
+{
+  "index_patterns": "*jaeger-service-*",
+  "order": 1,
+  "aliases": {
+    "jaeger-service-read": {}
+  },
+  "settings": {
+    "index.number_of_shards": ${__NUMBER_OF_SHARDS__},
+    "index.number_of_replicas": ${__NUMBER_OF_REPLICAS__},
+    "index.mapping.nested_fields.limit": 50,
+    "index.requests.cache.enable": true,
+    "lifecycle": {
+      "name": "jaeger",
```

Comment:
I didn't want to add a new set of index templates. 

Could we instead add the `lifecycle` configuration to our existing index templates?

Reply:
@pavolloffay  - I tried that initially, When jaeger comes up, it is also trying to create the index templates ( even with --es.use-alias enabled) and if the template name is same - it is overriding it. Hence had to use new set of templates with higher priority (1).

Reply:
Even if the templates are overridden it should not be an issue since the definition does not change it just references the name of ILM. 

`--es.create-index-templates` can be used to disable template creation

Reply:
lifecycle or ILM is inside settings of index template -

```"settings": {
    "index.number_of_shards": ${__NUMBER_OF_SHARDS__},
    "index.number_of_replicas": ${__NUMBER_OF_REPLICAS__},
    "index.mapping.nested_fields.limit": 50,
    "index.requests.cache.enable": true,
    "lifecycle": {
      "name": "jaeger",
      "rollover_alias": "${__INDEX_WRITE_ALIAS__}"
    }
```
Default template has settings without lifecycle. Hence when we start jaeger - it overrides the settings of the template ( unless --es.create-index-templates is used to disable it). That means we are saying user has to use --es.use-alias with --es.create-index-templates always - if ILM is needed.

Reply:
I get it now, I want to avoid having multiple variations of the index templates.

Can be the lifecycle be configured but a PUT (change) request?

Reply:
> I get it now, I want to avoid having multiple variations of the index templates.

Yes - completely agree. Ideal thing would be disable creation of templates from jaeger end whenever we have --es.use-aliases.

> Can be the lifecycle be configured but a PUT (change) request?
Unfortunately we have to run init before starting jaeger. And index-templates must be created before 1st index is created. So, right now, I see no way but to have another set of templates for ILM with different name and higher order - so that when index templates are applied - ILM template takes precedence.

Either this or we somehow mandate to always use --es.use-alias with --es.create-index-templates set to false.

Reply:
> Ideal thing would be to disable creation of templates from jaeger end whenever we have --es.use-aliases

@pavolloffay,  I think I know why this was not done. As of now, If we enable use-aliases and start jaeger with out running esRollover init first --> Jaeger is creating the default index templates and two initial **indices** with names jaeger-span-write & jaeger-span-read, so that it would continue to work.

Reply:
@pavolloffay  - As suggested - changes made to code to avoid using multiple variations of index templates. Please go through the latest commit

Reply:
@pavolloffay  Any feedback on this?

Reply:
@pavolloffay - Let me know, if you have any thoughts on the approach